modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,931 @@
+
+                if (_BoundaryNormal == 0)
+                {
+                  miubalibwR = ((miuow) * (qwR[a])) - ((miuox) * (qxR[a]) * (-1)) -
+                               ((miuoy) * (qyR[a]) * (-1)) - ((miuoz) * (qzR[a]) * (-1));
+                  miubalibxR = ((miuow) * (qxR[a]) * (-1)) + ((miuox) * (qwR[a])) +
+                               ((miuoy) * (qzR[a]) * (-1)) - ((miuoz) * (qyR[a]) * (-1));
+                  miubalibyR = ((miuow) * (qyR[a]) * (-1)) - ((miuox) * (qzR[a]) * (-1)) +
+                               ((miuoy) * (qwR[a])) + ((miuoz) * (qxR[a]) * (-1));
+                  miubalibzR = ((miuow) * (qzR[a]) * (-1)) + ((miuox) * (qyR[a]) * (-1)) -
+                               ((miuoy) * (qxR[a]) * (-1)) + ((miuoz) * (qwR[a]));
+                }
+
+                _VwlibR[_qp] = miubalibwR;
+                _VxlibR[_qp] = miubalibxR;
+                _VylibR[_qp] = miubalibyR;
+                _VzlibR[_qp] = miubalibzR;
+
+                if (sumval == 0.0)
+                {
+                  miubalibwR = 0.0;
+                  miubalibxR = 0.0;
+                  miubalibyR = 0.0;
+                  miubalibzR = 0.0;
+
+                  _VwlibR[_qp] = 0.0;
+                  _VxlibR[_qp] = 0.0;
+                  _VylibR[_qp] = 0.0;
+                  _VzlibR[_qp] = 0.0;
+                }
+
+                // dot product of normalized anisotropic vector and normalized gradient vector
+
+                ADReal dot_product =
+                    (miubalibxR * vsmallbax) + (miubalibyR * vsmallbay) + (miubalibzR * vsmallbaz);
+
+                // calculate the switch for a specific pair of grains and each library
+                // misorientation
+
+                if ((thetabalib[l]) < 0)
+                {
+                  (thetabalib[l]) = 6.28319 + (thetabalib[l]);
+                }
+
+                ADReal thetadtheta = thetaba - (thetabalib[l]);
+
+                ADReal normvbavalue = (vbax * vbax) + (vbay * vbay) + (vbaz * vbaz);
+
+                ADReal normvba = std::pow(normvbavalue, .5);
+
+                ADReal normvbalibvalue = (vbalibx[l] * vbalibx[l]) + (vbaliby[l] * vbaliby[l]) +
+                                         (vbalibz[l] * vbalibz[l]);
+
+                ADReal normvbalib = std::pow(normvbalibvalue, .5);
+
+                ADReal cosinevalue1 =
+                    ((vbalibx[l]) * vbax) + ((vbaliby[l]) * vbay) + ((vbalibz[l]) * vbaz);
+
+                ADReal cosinevalue2 = normvbalib * normvba;
+
+                ADReal cosinevalue3 = cosinevalue1 / cosinevalue2;
+
+                ADReal thetadv = std::acos(cosinevalue3);
+
+                ADReal switchvalue =
+                    (-100) * (((thetadv / _alphaswitch) * (thetadv / _alphaswitch)) +
+                              ((thetadtheta / _betaswitch) * (thetadtheta / _betaswitch)));
+
+                _S_switch[_qp] = std::exp(switchvalue);
+
+                if (_S_switch[_qp] < 0.001)
```

Comment:
What is `0.001`?

Reply:
That was fixed. See updated files. I added an explanation in the file.